plugin/trino-hive/src/main/java/io/trino/plugin/hive/metastore/thrift/ThriftHiveMetastore.java

```diff
@@ -1079,7 +1082,12 @@ public void dropTable(String databaseName, String tableName, boolean deleteData)
                     }));
         }
         catch (NoSuchObjectException e) {
-            throw new TableNotFoundException(new SchemaTableName(databaseName, tableName), e);
+            if (attemptCount.get() == 1) {
+                // Throw exception only on first attempt.
+                throw new TableNotFoundException(new SchemaTableName(databaseName, tableName), e);
+            }
+            // If table is not found on consecutive attempts it was probably dropped on first attempt and timeout occurred.
+            // Exception in such case can be safely ignored.
```

Comment:
There're 2 places NoSuchObjectException can happen - getTable & dropTable. This comment is true for getTable, but what about dropTable?

Does dropTable method throw the exception only for **_table_** and doesn't throw the same exception for other reasons (e.g. failed to drop partitions due to concurrent operation)?

Reply:
What about wrapping in an additional `try/catch`  the following line  to make sure that the handling is done exclusively for the `getTable` metastore call:

```
Table table = client.getTable(databaseName, tableName);
```